### DISCUSSION TITLE
How to make walls?

### DISCUSSION THREAD

**Discussion by vedrankolac** [enable3d/enable3d#273] [Help / Questions] [answered]
Hi! 👋🏼

What would be the right way to make walls in ammo?

I want to make walls that don't move when they are collided with other objects - I want them to be completely static.
Should I use the `physics.add.box()` or `physics.add.ground()` method?

**My issue**
When I create a wall with `physics.add.box()` and `collisionFlag = 2` the wall is still reacting/moving after being collided with other objects. Is there something else I should configure to make it not react/move on collisions?

What is the general practice in this case?

Thanks!
BLT

**Comment by yandeu** [comment, marked as answer]
`ground()` is a wrapper around `box()`.

I suggest you use `box()` with collisionFlags 1 (static) and a mass of 0.

See https://github.com/enable3d/enable3d/blob/af247fe4bb8de2d1ac4265162cc4abb57cf71dea/packages/ammoPhysics/src/shapes.ts#L46

**Reply by vedrankolac** [reply]
Using `box()`, setting mass to 0 and collisionFlags to 1 makes no difference, walls are still moving on collision.

````
  const wallLeft = physics.add.box({
    width: wallSize,
    height: floorSize + wallSize * 2,
    depth: floorSize + wallSize * 2,
    x: -floorSize/2 - wallSize/2,
    y: floorSize/2,
    z: 0,
  });
  wallLeft.visible = isVisible;
  wallLeft.body.setCollisionFlags(1);
  wallLeft.body.mass = 0;
```

**Reply by vitalspace** [reply]
EASY:

```
        const createWall = (x, y, z) => {
          // Create THREE BOX
          this.wall = this.add.box({ width:0, height: 10, depth: 10 });
          // Set Box Position
          this.wall.position.set(x, y, z);
          // Add Physics to the box three
          this.physics.add.existing(this.wall, {
            shape: 'mesh',
            mass: 0,
            collisionFlags: 1,
            autoCenter: false
          })
        }
        createWall(35, -4, 0);
```
https://user-images.githubusercontent.com/29004070/169896983-752bc7de-7c6f-459b-a575-8463f0d65255.mp4

**Reply by vedrankolac** [reply]
Great!
I'll try it tmrw :)
Thanks!!
